frontend/src/components/Topology/Topology.js

```diff
@@ -0,0 +1,273 @@
+            availableFilters: {},
+            activeFilters: {},
+            otherTypeFilters: [],
+            showChannelsControl: false,
+        }
+
+        // merge styles and options with defaults
+        this.ResourceFilterView = createRef()
+    }
+
+    UNSAFE_componentWillReceiveProps(nextProps) {
+        // Keep channel control visible if multiple channels exist, or hide it otherwise
+        this.setState({
+            showChannelsControl: _.get(nextProps.channelControl, 'allChannels', []).length > 1,
+        })
+
+        this.setState((prevState) => {
+            let { timestamp } = prevState
+            const { userIsFiltering } = prevState
+            const {
+                elements: { nodes },
+                fetchControl = {},
+            } = nextProps
+            const { isLoaded = true, isReloading = false } = fetchControl
+
+            if (!_.isEqual(nodes, this.props.elements.nodes) && !isReloading) {
+                timestamp = new Date().toString()
+            }
+            this.props.options.updateNodeStatus(nodes, this.props.t)
+            const { availableFilters, activeFilters, otherTypeFilters } = this.props.options.getAllFilters(
+                isLoaded,
+                nodes,
+                this.props.options,
+                prevState.activeFilters,
+                this.knownTypes,
+                userIsFiltering,
+                this.props.t
+            )
+            return {
+                links: _.cloneDeep(nextProps.elements.links),
+                nodes: _.cloneDeep(nextProps.elements.nodes),
+                isLoaded,
+                timestamp,
+                availableFilters,
+                activeFilters,
+                otherTypeFilters,
+            }
+        })
+    }
+
+    shouldComponentUpdate(nextProps, nextState) {
+        return (
+            !_.isEqual(this.state.nodes, nextState.nodes) ||
+            !_.isEqual(this.state.links, nextState.links) ||
+            !_.isEqual(this.props.fetchControl, nextProps.fetchControl) ||
+            !_.isEqual(
+                _.omit(this.props.channelControl, _.functions(this.props.channelControl)),
+                _.omit(nextProps.channelControl, _.functions(nextProps.channelControl))
+            ) ||
+            !_.isEqual(this.state.availableFilters, nextState.availableFilters) ||
+            !_.isEqual(this.state.activeFilters, nextState.activeFilters) ||
+            this.props.canUpdateStatuses !== nextProps.canUpdateStatuses ||
+            this.state.searchName !== nextState.searchName ||
+            this.state.fetchChannel !== nextState.fetchChannel ||
+            this.state.isLoaded !== nextState.isLoaded
+        )
+    }
+
+    render() {
+        const { fetchControl = {}, t } = this.props
+        const { isFailed = false } = fetchControl
+
+        if (isFailed) {
+            return <AcmAlert className="persistent" title={t('error.default.description')} variant="danger" isInline />
```

Comment:
Because of the translation, should we do t('the text') instead of using the .json files?

Reply:
We do have a mix currently. It's not required at the moment to do it one way or the other.

Reply:
kk sounds good!